app/controllers/api/v2/users_controller.rb

```diff
@@ -118,6 +118,11 @@ def destroy
         end
       end
 
+      api :GET, "/menu", N_("Get menu items for current user")
+      def menu
+        render :json => UserMenu.new.generate
+      end
+
```

Comment:
I'm not too keen on adding this to the public api since these tend to be a pain to change if we want to later on (e.g. see the effort we need to get rid of endpoints/parameters we deprecated years ago). 
Would it make sense to make this part of the UI controller instead? that way perhaps in the future the same endpoint can feed both the UI itself (which now loads this info from server-side generated helper) and the tests?